operator/webserver.py

```diff
@@ -41,12 +67,50 @@ def mission_stop():
             # wait for it to finish
             interfaceprocess.wait()
 
-    return {'message': 'Mission stopped'}
+    car_controls.brake = 1
+    client.setCarControls(car_controls)
+    car_controls.brake = 0 #remove brake
+
+    log = '{}: {}'.format(str(datetime.now()), 'Mission stopped.')
+    logs.append(log)
+    del logs[:]
+
+    log_file.write(log + '\n')
+    log_file.close()
+
+    return {'response': log}
 
 @app.route('/mission/reset', methods=['POST'])
 def mission_reset():
     client.reset()
-    return {'message': 'Car reset'}
+
+    log = '{}: {}'.format(str(datetime.now()), 'Car reset.')
+    logs.append(log)
+    log_file.write(log + '\n')
+
+    return {'response': log}
+
+@app.route('/logs', methods=['GET'])
+def get_logs():
+    return {'response': logs}
+
+def referee_state_listener():
+    Timer(2.0, referee_state_listener).start()
+    ref = client.getRefereeState()
+    global doo_count, laps
+
+    if doo_count != ref.doo_counter:
+        doo_count = ref.doo_counter
+        log = '{}: {}. {} {}'.format(str(datetime.now()), 'Cone hit', doo_count, 'DOO cone(s).')
+        logs.append(log)
+        log_file.write(log + '\n')
+
+    if laps != ref.laps:
+        laps = ref.laps
+        log = '{}: {} {} {}'.format(str(datetime.now()), 'Lap completed. Completed', laps, 'lap(s).')
```

Comment:
can we add `(seconds)` somewhere?

can we round the lap times to 3 decimals after the comma?

![Screenshot from 2020-05-24 01-26-29](https://user-images.githubusercontent.com/8008227/82742428-40ed4680-9d5e-11ea-931f-2ed4eab7c014.png)

Reply:
Oh lol. I thought this returned the lap number. Did not test properly

Reply:
Fixed